fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneStoredFieldsTest.java

```diff
@@ -0,0 +1,395 @@
+        }
+    }
+
+    @ParameterizedTest
+    @BooleanSource
+    void testInsertUpdateDocuments(boolean useOptimizedStoredFieldsFormat) throws Exception {
+        Index index = getSimpleTextIndex(useOptimizedStoredFieldsFormat);
+
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, "Document 1", 2));
+            recordStore.saveRecord(createSimpleDocument(1624L, "Document 2", 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, "NonDocument 3", 2));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.updateRecord(createSimpleDocument(1623L, "Document 3 modified", 2));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            final RecordQuery query = buildQuery("Document", Collections.emptyList(), SIMPLE_DOC);
+            queryAndAssertFields(query, "text", Set.of(1623L, 1624L), Set.of("Document 2", "Document 3 modified"));
+        }
+    }
+
+    @ParameterizedTest
+    @BooleanSource
+    void testDeleteAllDocuments(boolean useOptimizedStoredFieldsFormat) throws Exception {
+        Index index = getSimpleTextIndex(useOptimizedStoredFieldsFormat);
+
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, "Document 1", 2));
+            recordStore.saveRecord(createSimpleDocument(1624L, "Document 2", 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, "NonDocument 3", 2));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.deleteRecord(Tuple.from(1623L));
+            recordStore.deleteRecord(Tuple.from(1624L));
+            recordStore.deleteRecord(Tuple.from(1547L));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            final RecordQuery query = buildQuery("Document", Collections.emptyList(), SIMPLE_DOC);
+            queryAndAssertFields(query, "text", Set.of(), Set.of());
+            if (useOptimizedStoredFieldsFormat) {
+                try (FDBDirectory directory = new FDBDirectory(recordStore.indexSubspace(index), context, index.getOptions())) {
+                    // When deleting all docs from the index, the last segment (_1) gets removed, leaving the _0 segment with the 3 tombstones
+                    assertDocCountPerSegment(directory, List.of("_0", "_1"), List.of(3, 0));
```

Comment:
If you run a merge like in: https://github.com/FoundationDB/fdb-record-layer/pull/2364/files#diff-eb73467a288b09c88277b4ba28abdf2e1dc24cbae14e25808c479ed69aba6ec8R1306
You should be able to re-scan and see that there are no documents, as `_0` will be deleted.

Reply:
I'm not seeing the diff. Do you mean like:
```
        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
                 .setRecordStore(recordStore)
                 .setIndex(index)
                 .build()) {
             indexBuilder.mergeIndex();
         }
```

Reply:
Yes, that is what it was linking to, the lines that run the merge.

Reply:
I tried running this after multiple commits and it didn't reduce the segment count. Added a test with a TODO to show the issue. I haven't looked at the other tests that used this snippet to see what is happening though.